cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1082,7 +1095,12 @@ class _ComputeResourceSchema(BaseSchema):
 class SlurmComputeResourceSchema(_ComputeResourceSchema):
     """Represent the schema of the Slurm ComputeResource."""
 
-    instance_type = fields.Str(required=True, metadata={"update_policy": UpdatePolicy.COMPUTE_FLEET_STOP})
+    instance_type = fields.Str(metadata={"update_policy": UpdatePolicy.COMPUTE_FLEET_STOP})
+    instance_types = fields.Nested(
+        FlexibleInstanceTypeSchema,
```

Comment:
What about going for "InstanceTypeSchema"? I think it's simpler

Reply:
I initially used `InstanceTypeSchema` but figured it might be confused with the properties in an InstanceTypeInfo object.
But since this schema is used in the context of a FlexibleComputeResource, I can simple use `InstanceTypeSchema`.